File: backend-api/reservation-api/src/main/java/com/hcmute/reservation/model/dto/table/FloorMapTableResponse.java
package com.hcmute.reservation.model.dto.table;

import com.hcmute.reservation.model.enums.ReservationStatus;
import com.hcmute.reservation.model.enums.TableStatus;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class FloorMapTableResponse {
    private Long tableId;
    private int capacity;
    private TableStatus status;
    private boolean isActive;
    // null nếu không có đơn
    private Long currentReservationId;
    private String currentCustomerName;
    private ReservationStatus currentReservationStatus;
}

File: desktop-staff/reservation-winforms/DTO/table/FloorMapTableResponse.cs
using Newtonsoft.Json;
using System;

namespace reservation_winforms.DTO.table
{
    public class FloorMapTableResponse
    {
        [JsonProperty("tableId")]
        public long TableId { get; set; }

        [JsonProperty("capacity")]
        public int Capacity { get; set; }

        [JsonProperty("status")]
        public string Status { get; set; }

        [JsonProperty("active")]
        public bool IsActive { get; set; }

        [JsonProperty("currentReservationId")]
        public long? CurrentReservationId { get; set; }

        [JsonProperty("currentCustomerName")]
        public string CurrentCustomerName { get; set; }

        [JsonProperty("currentReservationStatus")]
        public string CurrentReservationStatus { get; set; }

        [JsonProperty("currentReservationTime")]
        public DateTime? CurrentReservationTime { get; set; }
    }
}